packages/shared/lib/sdk/sync.ts

```diff
@@ -418,10 +420,14 @@ export class NangoAction {
         return this.nango.getToken(this.providerConfigKey as string, this.connectionId as string);
     }
 
-    public async getConnection(): Promise<Connection> {
+    public async getConnection(optionalProviderConfigKey?: string, optionalConnectionId?: string): Promise<Connection> {
         this.exitSyncIfAborted();
+
+        const providerConfigKey = optionalProviderConfigKey || this.providerConfigKey;
+        const connectionId = optionalConnectionId || this.connectionId;
+
         if (!this.memoizedConnection || Date.now() - this.memoizedConnection.timestamp > MEMOIZED_CONNECTION_TTL) {
```

Comment:
are you sure it doesn't break memoizedConnection? 
passing a parameter does not change the fact that the connection can be memoized and the wrong one will be returned no?